cpp/src/arrow/engine/CMakeLists.txt

```
@@ -48,8 +48,7 @@ set(SUBSTRAIT_PROTOS
     type_expressions)
 
 externalproject_add(substrait_ep
-                    GIT_REPOSITORY "${ARROW_SUBSTRAIT_REPO}"
-                    GIT_TAG "${ARROW_SUBSTRAIT_TAG}"
+                    URL "https://github.com/${ARROW_SUBSTRAIT_REPO}/archive/${ARROW_SUBSTRAIT_TAG}.tar.gz"
```

```suggestion
                    URL "${ARROW_SUBSTRAIT_REPO}/archive/${ARROW_SUBSTRAIT_TAG}.tar.gz"
```

(the https://github.com/ is included in the variable)

---

This url will only work for github so we my want to restore the previous url and note that `ARROW_SUBSTRAIT_REPO` should be a github `user/repo` pair.